independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/interceptors/bindings/repeatable/InheritedStereotypeWithRepeatableInterceptorBindingTest.java

```diff
@@ -0,0 +1,121 @@
+package io.quarkus.arc.test.interceptors.bindings.repeatable;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.test.ArcTestContainer;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import javax.enterprise.inject.Stereotype;
+import javax.inject.Singleton;
+import javax.interceptor.AroundInvoke;
+import javax.interceptor.Interceptor;
+import javax.interceptor.InterceptorBinding;
+import javax.interceptor.InvocationContext;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+/**
+ * Tests usage of inherited stereotype containing a repeating interceptor binding.
+ */
+public class InheritedStereotypeWithRepeatableInterceptorBindingTest {
+    @RegisterExtension
+    public ArcTestContainer container = new ArcTestContainer(MyBinding.class, MyBinding.List.class, MyStereotype.class,
+            MyBeanDefiningAnnotation.class, MySuperclass.class, MyBean.class, IncrementingInterceptor.class);
+
+    @Test
+    @Disabled("ArC doesn't have stereotype inheritance yet")
+    public void test() {
+        MyBean bean = Arc.container().instance(MyBean.class).get();
+
+        assertEquals(10, bean.foo());
+        assertEquals(21, bean.foobar());
+        assertEquals(30, bean.foobaz());
+        assertEquals(41, bean.foobarbaz());
+        assertEquals(51, bean.nonannotated());
+    }
+
+    @Target({ ElementType.TYPE, ElementType.METHOD, ElementType.CONSTRUCTOR })
+    @Retention(RetentionPolicy.RUNTIME)
+    @Repeatable(MyBinding.List.class)
+    @InterceptorBinding
+    @interface MyBinding {
+        String value();
+
+        @Target({ ElementType.TYPE, ElementType.METHOD, ElementType.CONSTRUCTOR })
+        @Retention(RetentionPolicy.RUNTIME)
+        @interface List {
+            MyBinding[] value();
+        }
+    }
+
+    @MyBinding("foo")
+    @MyBinding("bar")
```

Comment:
Hm, I don't think this should be supported. A stereotype with interceptor bindings is inherently a class-level-only concept and so I would expect the same limitations as with bindings declared on a class...

Reply:
The stereotype is class-only.

The test also has methods with the same interceptor binding, verifying that if a method has an interceptor binding, other interceptor bindings (from the class, from superclasses, from stereotype) don't apply, but that's a valid case IMHO?

Reply:
(Ah and this test is `@Disabled`, because we don't do stereotype inheritance yet. But it's still valid to discuss I believe.)

Reply:
My point is that a stereotype with multiple bindings of the same type should probably result in a definition exception...